src/core/geometry/qgstriangle.cpp

```diff
@@ -373,9 +373,9 @@ QVector<double> QgsTriangle::lengths() const
   if ( isEmpty() )
     return lengths;
 
-  lengths.append( vertexAt( 0 ).distance( vertexAt( 1 ) ) );
-  lengths.append( vertexAt( 1 ).distance( vertexAt( 2 ) ) );
-  lengths.append( vertexAt( 2 ).distance( vertexAt( 0 ) ) );
+  lengths.append( vertexAt( 1 ).distance( vertexAt( 2 ) ) ); // a = |BC|
+  lengths.append( vertexAt( 0 ).distance( vertexAt( 2 ) ) ); // b = |AC|
+  lengths.append( vertexAt( 0 ).distance( vertexAt( 1 ) ) ); // c = |AB|
```

Comment:
This will break the current API in which the behaviour is precisely  defined as:
https://qgis.org/api/classQgsTriangle.html#ae4e5d068c5572b9a41c4081a84454215

> Lengths of triangle ABC where [AB] is at 0, [BC] is at 1, [CA] is at 2.

Maybe, if not strictly necessary, this change could be avoided, since it seems to me the `lengths()` function already returns correct results based on such definition in the API.

Reply:
You're right, fixed.